src/apple/code-tools.ts

```diff
@@ -0,0 +1,77 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as templates from './templates';
+
+const swiftAppLaunchRegex = /(func\s+application\s*\(_\sapplication:[^,]+,\s*didFinishLaunchingWithOptions[^,]+:[^)]+\)\s+->\s+Bool\s+{)|(init\s*\([^)]*\)\s*{)/gim;
+const objcAppLaunchRegex = /-\s*\(BOOL\)\s*application:\s*\(UIApplication\s*\*\)\s*application\s+didFinishLaunchingWithOptions:\s*\(NSDictionary\s*\*\)\s*launchOptions\s*{/gim;
+
+
+function isAppDelegateFile(filePath: string): boolean {
+    const appLaunchRegex = filePath.toLowerCase().endsWith(".swift") ? swiftAppLaunchRegex : objcAppLaunchRegex;
+
+    const fileContent = fs.readFileSync(filePath, 'utf8');
+    return (fileContent.includes("UIApplicationDelegate") && appLaunchRegex.test(fileContent)) || /struct[^:]+:\s*App\s*{/.test(fileContent);
```

Comment:
`m`:  Will this also work for Objective-C? Cause the `UIApplicationDelegate` is in the header file.

Reply:
Yes, we will, and thats true.
Thanks!!